File: 백준/Silver/1475. 방 번호/방 번호.c
#include <stdio.h>

int max(int arr[], int length){
   int max = -1;
   for(int i = 0; i < length; i++){
      if (arr[i] > max){
         max = arr[i];
      }
   }
   return max;
}

int main(void) {  
   int arr[10] = {0};
   int num;
   scanf("%d", &num);

   int count;
   while(num > 0) {
      count = num % 10;

      if(arr[6] < arr[9] && (count == 6 || count == 9)){
         arr[6] += 1;
      }
      else if(arr[6] > arr[9] && (count == 6 || count == 9)){
         arr[9] += 1;
      }
      else{
         arr[count] += 1;
      }
      num = num / 10;
   }
   printf("%d ", max(arr, 10));
   
   return 0;
}

File: 백준/Silver/1475. 방 번호/방 번호.py
arr = [0 for _ in range(10)]
num = list((input()))
length = len(num)
for i in range(length):
  count = int(num[i])
  if (count == 6 or count == 9) and arr[6] < arr[9]:
    arr[6] += 1
  elif (count == 6 or count == 9) and arr[6] > arr[9]:
    arr[9] += 1
  else:
    arr[count] += 1

print(max(arr))